toolkit/tools/grapher/grapher.go

```diff
@@ -205,6 +208,36 @@ func addNodesForPackage(g *pkggraph.PkgGraph, pkg *pkgjson.Package) (foundDuplic
 	return
 }
 
+// handleRemoteDependency ensures that a remote node is available in the graph for every unresolved dependency.
+// 1. Check if the exact dependency is already in the graph. If it is, reuse it.
+// 2. If it is not, create a new unresolved node for the dependency.
+// It is important that we only match on the exact dependency name and version. If we don't, we may end up with
+// unpredictable behavior in the scheduler. If two different remote dependencies are added to two different build
+// nodes of a single SRPM, then the scheduler may queue that node twice.
+func handleRemoteDependency(g *pkggraph.PkgGraph, dependency *pkgjson.PackageVer) (resolvedNode *pkggraph.PkgNode, err error) {
```

Comment:
Naming suggestion: "handle" feel rather vague. Since we're already making a distinction between "Find**Best**" and "Find**Exact**" and we're calling `handleRemoteDependency` from inside `addSingleDependency`, what do you think about calling `handleRemoteDependency` `addExactRemoteDependency`. This way it's clearer that the effect of the function is an optional addition of a new node and that it's a subset of `addSingleDependency`, because we're focusing explicitly on exact-matched remote dependencies.